From cceac8459bda753b1709499f47606d83a47c1452 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C4=90o=C3=A0n=20Tr=E1=BA=A7n=20C=C3=B4ng=20Danh?=
 <congdanhqx@gmail.com>
Date: Sat, 31 Oct 2020 19:41:43 +0700
Subject: [PATCH] Don't use thread safety annotations with gcc

These annotations are primarily useful for detecting race conditions in
code. They don't seem to have any runtime effect and are not supported
by gcc.

See: https://clang.llvm.org/docs/ThreadSafetyAnalysis.html
---
 include/android-base/thread_annotations.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/android-base/thread_annotations.h b/include/android-base/thread_annotations.h
index 53fe6da..bc62f80 100644
--- a/include/android-base/thread_annotations.h
+++ b/include/android-base/thread_annotations.h
@@ -18,7 +18,11 @@
 
 #include <mutex>
 
+#ifdef __clang__
 #define THREAD_ANNOTATION_ATTRIBUTE__(x) __attribute__((x))
+#else
+#define THREAD_ANNOTATION_ATTRIBUTE__(x)
+#endif
 
 #define CAPABILITY(x) \
       THREAD_ANNOTATION_ATTRIBUTE__(capability(x))
